core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -510,11 +510,18 @@ public List<QualifiedObjectName> listTables(Session session, QualifiedTablePrefi
 
         Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
         if (objectName.isPresent()) {
-            Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
-            if (relationType.isPresent()) {
-                return ImmutableList.of(objectName.get());
+            try {
+                Optional<RelationType> relationType = getRelationTypeIfExists(session, objectName.get());
+                if (relationType.isPresent()) {
+                    return ImmutableList.of(objectName.get());
+                }
+                // TODO we can probably return empty lit here
+            }
+            catch (RuntimeException e) {
+                handleListingError(e, prefix);
+                // Empty in case of metadata error.
+                return ImmutableList.of();
```

Comment:
I think we should return `ImmutableList.of(objectName.get())` in some situations. 

For instance, when "Table SerdeInfo is null", it means the table actually exists. Returning empty list doesn't make sense in this situation. 

Does this worth leaving TODO comment?